File: Actividad 3 ParadigmasProg/act3Ruby/conversorMedidas/conversorMedidas.rb
  puts "Que quiere convertir?"
  puts "1)Milimetros 2)Centimetros 3)Metros 4)Kilometros"
  print "Opcion: "
  opci = gets.chomp.to_i

    case opci
    when 1
      puts "1)Milimetros a centimetros"
      puts "2)Mililmetros a metros"
      puts "3)Milimetros a kilometros"
      print "Opcion: "
      opcion = gets.chomp.to_i
      print "Milimetros: "
      medida = gets.chomp.to_f

      case opcion
      when 1
        medida_final = medida / 10
      when 2
        medida_final = medida / 1000
      when 3
        medida_final = medida / 1000000
      else
        puts "Opcion no valida"
        return
      end

    when 2
      puts "1)Centimetros a milimetros"
      puts "2)Centimetros a metros"
      puts "3)Centimetros a kilometros"
      print "Opcion: "
      opcion = gets.chomp.to_i
      print "Centimetros: "
      medida = gets.chomp.to_f

      case opcion
      when 1
        medida_final = medida * 10
      when 2
        medida_final = medida / 100
      when 3
        medida_final = medida / 100000
      else
        puts "Opcion no valida"
        return
      end

    when 3
      puts "1)Metros a milimetros"
      puts "2)Metros a centimetros"
      puts "3)Metros a kilometros"
      print "Opcion: "
      opcion = gets.chomp.to_i
      print "Metros: "
      medida = gets.chomp.to_f

      case opcion
      when 1
        medida_final = medida * 1000
      when 2
        medida_final = medida * 100
      when 3
        medida_final = medida / 1000
      else
        puts "Opcion no valida"
        return
      end

    when 4
      puts "1)Kilometros a milimetros"
      puts "2)Kilometros a centimetros"
      puts "3)Kilometros a metros"
      print "Opcion: "
      opcion = gets.chomp.to_i
      print "Kilometros: "
      medida = gets.chomp.to_f

      case opcion
      when 1
        medida_final = medida * 1000000
      when 2
        medida_final = medida * 100000
      when 3
        medida_final = medida * 1000
      else
        puts "Opcion no valida"
        return
      end

    else
      puts "Opcion no valida"
      return
    end

    puts "Resultado: #{medida_final}"

File: Ejercicio clase 18042024/ParadigmasProg/src/ConversorMedidas.java
import java.util.Scanner;

public class ConversorMedidas 
{
    public static void main(String[] args) 
    {
        Scanner leer = new Scanner(System.in);
        int opci, opcion;
        float medida, medidaFinal;

        System.out.print("Que quiere convertir? \n1)Milimetros 2)centimetros 3)Metros 4)Kilometros \nOpcion: ");
        opci = leer.nextInt();

        switch (opci) 
        {
            case 1:
                System.out.print("1)Milimetros a centimetros \n2)Milimetros a metros \n3)Milimetros a kilometros \nOpcion:  ");
                opcion = leer.nextInt();
                System.out.print("Milimetros: ");
                medida = leer.nextFloat();
                switch (opcion) 
                {
                    case 1:
                        medidaFinal = medida / 10;
                        break;
                    case 2:
                        medidaFinal = medida / 1000;
                        break;
                    case 3:
                        medidaFinal = medida / 1000000;
                        break;
                    default:
                        System.out.println("Opción no válida");
                        return;
                }
                break;
            case 2:
                System.out.print("1)Centimetros a milimetros \n2)Centimetros a metros \n3)Centimetros a kilometros \nOpcion:  ");
                opcion = leer.nextInt();
                System.out.print("Centimetros: ");
                medida = leer.nextFloat();
                switch (opcion) 
                {
                    case 1:
                        medidaFinal = medida * 10;
                        break;
                    case 2:
                        medidaFinal = medida / 100;
                        break;
                    case 3:
                        medidaFinal = medida / 100000;
                        break;
                    default:
                        System.out.println("Opción no válida");
                        return;
                }
                break;
            case 3:
                System.out.print("1)Metros a milimetros  \n2)Metros a centimetros \n3)Metros a kilometros \nOpcion:  ");
                opcion = leer.nextInt();
                System.out.print("Metros: ");
                medida = leer.nextFloat();
                switch (opcion) 
                {
                    case 1:
                        medidaFinal = medida * 1000;
                        break;
                    case 2:
                        medidaFinal = medida * 100;
                        break;
                    case 3:
                        medidaFinal = medida / 1000;
                        break;
                    default:
                        System.out.println("Opción invalida");
                        return;
                }
                break;
            case 4:
                System.out.print("1)Kilometros a milimetros \n2)Kilometros a centimetros \n3)Kilometros a metros \nOpcion:  ");
                opcion = leer.nextInt();
                System.out.print("Kilometros: ");
                medida = leer.nextFloat();
                switch (opcion) 
                {
                    case 1:
                        medidaFinal = medida * 1000000;
                        break;
                    case 2:
                        medidaFinal = medida * 100000;
                        break;
                    case 3:
                        medidaFinal = medida * 1000;
                        break;
                    default:
                        System.out.println("Opción invalida");
                        return;
                }
                break;
            default:
                System.out.println("Opción invalida");
                return;
        }
        System.out.println("Resultado: " + medidaFinal);
    }
}